src/nsntrace.c

```diff
@@ -245,7 +253,9 @@ netns_main(void *arg) {
 	} else if (child_pid > 0) { /* parent - tracer */
 		struct timespec timeout = { 0 };
 
-		waitpid(child_pid, NULL, 0);
+		waitpid(child_pid, &status, 0);
+		ret = WIFEXITED(status) ? WEXITSTATUS(status) : EXIT_FAILURE;
```

Comment:
I can fix it while merging!

Thanks a lot for the nice patch series!

2016-08-05 16:33 GMT+02:00 Chris Lamb <notifications@github.com>:

> In src/nsntrace.c
> <https://github.com/jonasdn/nsntrace/pull/10#discussion_r73701867>:
>
> > @@ -245,7 +253,9 @@ netns_main(void *arg) {
> >  	} else if (child_pid > 0) { /* parent - tracer */
> >  		struct timespec timeout = { 0 };
> >
> > -		waitpid(child_pid, NULL, 0);
> > +		waitpid(child_pid, &status, 0);
> > +		ret = WIFEXITED(status) ? WEXITSTATUS(status) : EXIT_FAILURE;
>
> No strong feelings here. Would you like an updated patch series or are you
> happy changing it post-merge?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jonasdn/nsntrace/pull/10/files/cceb803971c9be4ca4a44ef443646f5d3f3ac856#r73701867>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA7eQe9unlT72jPn4xp93hFGhS4l6j2iks5qc0m2gaJpZM4JdkLQ>
> .
>